bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/HeapSizeManager.java

```diff
@@ -64,60 +74,73 @@ public long getMaxHeapSize() {
   }
 
   public synchronized void waitUntilAllOperationsAreDone() throws InterruptedException {
+    isFlushing.set(true);
     boolean performedWarning = false;
-    while(!pendingOperationsWithSize.isEmpty()) {
+    while (true) {
```

Comment:
Can we do this?

while (pendingOperationsWithSize.isEmpty()) {
  cleanupFinishedOperations();
  if (!performedWarning && requiresWarning()) {
    warn();
    performedWarning = true;
  }
  wait(FINISH_WAIT_MILLIS);
}